### ISSUE TITLE
Project not opening

### ISSUE THREAD

**Issue by daleinen7** [Scirra/Construct-3-bugs#1904]
## Problem description

Project now says "Failed to open project. Check it is a valid Construct 3 folder project."

## Attach a .c3p

https://drive.google.com/file/d/1ExoB3iXmwdN1WyZ3lnMysnq9j1RYwnag/view?usp=sharing

## Steps to reproduce

https://drive.google.com/open?id=1WREyYP_ET0vHUTlKzNdtrUsDbkmpOadl
Go to project bar, open the "Families" folder, open the "ShopOwners" family and remove "Penance" from the family. Save, close and reopen. 

## Observed result

On the website and the desktop .exe It says to check it is a valid file. The Windows Store app would sometimes give the same result and sometimes just crash giving a navigation error. 

## Expected result

I have the project under version control and none of the other versions open either. Also none of the old releases I tried would open it. This makes me think it has nothing to do with the newest update, but I couldn't get it to open in 112 but it did open last in 111. 

## Affected browsers

* Chrome: yes
* Desktop from Forum Link: yes
* Windows Store App: yes

both C3P and C3proj files

**Comment by AshleyScirra** [comment]
This looks like the same issue as #1821. There are various images in the project that appear to have been resized outside of Construct, so it thinks they're the wrong size. There's also a reference to a missing instance variable 'shopID'.

Construct should never save a project like this. I strongly suspect, as with the other issue, your source control has corrupted the project by a bad merge or something similar. However we did not action the other report because the reporter was unable to provide steps to reproduce the issue to take a working project and produce a bad project. Can you provide steps to follow to do this? We need a way to observe the problem happening in the first place before we can investigate further.

**Comment by daleinen7** [comment]
I'll see if I can reproduce it. I didn't edit any images outside of Construct but I did do a lot of replacing of images. And when committing I got an error I didn't understand. 

One weird thing is that when I checkout an older commit, I get the same corrupt project error in the Windows 10 store app, but not the desktop version from the forums. 

But it shouldn't be too hard to redo everything I did from the previous commit and see if I run into the same problem. I'll try to get to it tonight or tomorrow.

**Comment by AshleyScirra** [comment]
> I didn't edit any images outside of Construct but I did do a lot of replacing of images.
That's what I mean. Currently if you replace an image file in a project folder with a different-size one outside of Construct, it will corrupt the project.

**Comment by AshleyScirra** [comment]
Just checked and if we make Construct ignore the saved image sizes and decode images while loading so it can get the new size in case the image file got replaced, then projects take about 35% longer to load. So this seems to be a pretty unfortunate situation where people are doing something that breaks an optimisation.

**Comment by AshleyScirra** [comment]
Thanks to @shortercode we now have a faster way to read image sizes while loading projects - only ~11% slower rather than ~33% slower. I think this is a pretty good compromise to ensure projects load fast while not trusting the image size saved to JSON. Hopefully this will prevent changing image sizes outside of Construct causing issues loading/previewing projects. Let's see how this pans out in the next release.

This still doesn't do anything about the invalid 'shopID' variable reference though. I think that can be attributed to bad SVN merging though based on this:

> And when committing I got an error I didn't understand

**Comment by daleinen7** [comment]
I only did resizing in the Construct editor. I exported images at one size from the C3 editor, then opened a different sprite and imported them again with the editor. 

But it doesn't seem like it's the image sizes at all. It seems to have to do with families. I updated the "Steps to Reproduce" section, but here is a working project;

https://drive.google.com/open?id=1WREyYP_ET0vHUTlKzNdtrUsDbkmpOadl

Sorry the project is a bit cluttered, but if you go to the "Families" folder and open "ShopOwners" and remove "Penance" from the family and save, that will make the project not open on my computer. I hope this is enough to figure out why it's not working. That is the sprite with the "ShopID" instance variable that seems to be causing a problem. But surely someone has removed a sprite from a family with an instance variable in the past? Maybe because I'm calling the instance variable somewhere else?
